src/bdc/steps/step.py

```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Lucca Baumgärtner <lucca.baumgaertner@fau.de>
+
+
+class Step:
+    def __init__(self):
+        self._df = None
+        self._input_location = None
+        self._output_location = None
+
+    @property
+    def df(self):
+        return self._df
+
+    @df.setter
+    def df(self, df):
+        self._df = df
+
+    @property
+    def input_location(self):
+        return self._input_location
+
+    @input_location.setter
+    def input_location(self, input_location):
+        """
+        Set the location for input data. This could be an API endpoint or a CSV file path.
+        :param input_location: String file path or API endpoint
+        :return:
+        """
+        self._input_location = input_location
+
+    @property
+    def output_location(self):
+        return self._output_location
+
+    @output_location.setter
+    def output_location(self, output_location):
+        """
+        Set the location for output data. This could be an API endpoint or a CSV file path.
+        :param output_location: String file path or API endpoint
+        :return:
+        """
+        self._output_location = output_location
```

Comment:
Just to know why use getters and setters?

Reply:
It's more OOP style to not access class members from outside directly. But `{input|output}_location` where removed anyway.